examples/lib/tour/07-resource-management/example.go

```diff
@@ -0,0 +1,259 @@
+	// The context bundles all registrations and
+	// configuration settings, like credentials,
+	// which should be used when working with the OCM
+	// ecosystem.
+	ctx := ocm.DefaultContext()
+
+	// --- begin decouple ---
+	// gathering resources, this is completely hidden
+	// behind an implementation.
+	resources, err := GatherResources(ctx, CachingFactory{})
+	if err != nil {
+		return err
+	}
+
+	var list errors.ErrorList
+
+	list.Add(HandleResources(resources))
+
+	// we are done, so close the resources, again.
+	for i, r := range resources {
+		list.Addf(nil, r.Close(), "closing resource %d", i)
+	}
+	return list.Result()
+	// --- end decouple ---
+}
+
+// --- begin handle ---
+func HandleResources(resources []Resource) error {
+	var list errors.ErrorList
+	fmt.Printf("*** resources:\n")
+	for i, r := range resources {
+		fmt.Printf("  %2d: extra identity: %s\n", i+1, r.GetIdentity())
+		fmt.Printf("      resource type:  %s\n", r.GetType())
+		fmt.Printf("      access:         %s\n", r.GetAccess())
+	}
+
+	return list.Result()
+}
+
+// --- end handle ---
+
+func GatherResources(ctx ocm.Context, factory ResourceFactory) ([]Resource, error) {
+	var resources []Resource
+
+	spec := ocireg.NewRepositorySpec("ghcr.io/open-component-model/ocm")
+
+	// And the context can now be used to map the descriptor
+	// into a repository object, which then provides access
+	// to the OCM elements stored in this repository.
+	// --- begin repository ---
+	repo, err := ctx.RepositoryForSpec(spec)
+	if err != nil {
+		return nil, errors.Wrapf(err, "cannot setup repository")
+	}
+
+	// to release potentially allocated temporary resources,
+	// many objects must be closed, if they should not be used
+	// anymore.
+	// This is typically done by a `defer` statement placed after a
+	// successful object retrieval.
+	defer repo.Close()
+	// --- end repository ---
+
+	// Now, we look up the OCM CLI component.
+	// All kinds of repositories, regardless of their type
+	// feature the same interface to work with OCM content.
+	// --- begin lookup component ---
+	c, err := repo.LookupComponent("ocm.software/ocmcli")
+	if err != nil {
+		return nil, errors.Wrapf(err, "cannot lookup component")
+	}
+	defer c.Close()
+	// --- end lookup component ---
+
+	// Now we look for the versions of the component
+	// available in this repository.
+	versions, err := c.ListVersions()
+	if err != nil {
+		return nil, errors.Wrapf(err, "cannot query version names")
+	}
+
+	// OCM version names must follow the SemVer rules.
+	// Therefore, we can simply order the versions and print them.
+	err = semverutils.SortVersions(versions)
+	if err != nil {
+		return nil, errors.Wrapf(err, "cannot sort versions")
+	}
+	fmt.Printf("versions for component ocm.software/ocmcli: %s\n", strings.Join(versions, ", "))
+
+	// Now, we have a look at the latest version. it is
+	// the last one in the list.
+	// --- begin lookup version ---
+	cv, err := c.LookupVersion(versions[len(versions)-1])
```

Comment:
I have trouble accepting this code because it will require another make generate call on every release. Could we make this a static example so that it doesnt need to be updated constantly?

Reply:
I think, this would be another PR.